lib/format_routing_constraint.rb

```diff
@@ -0,0 +1,23 @@
+class FormatRoutingConstraint
+  def initialize(format, artefact_retriever = PreconfiguredArtefactRetriever)
```

Comment:
You're absolutely right, my thinking was a bit muddy here as I was imagining something quite different when I originally built the thing.

I've replaced the `PreconfiguredArtefactRetriever` class with an `ArtefactRetrieverFactory` static class which simply configures and returns an `ArtefactRetriever`. 

Thanks for the nudge!